docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+
+For example, control comments in this scenario could look like:
+
+```js
+//# sourceMappingURL=http://vpn-only-server.com:3000/PATH/TO/app1.js.map
+//# sourceMappingURL=http://authenticated-server.com:3000/PATH/TO/app1.js.map
+```
+
+The application build process should then go through these steps:
+
+1. On building the application, publish the map files to a private server and
+   capture the published URL of each map file.
+2. Ensure that the build process writes a correct `sourceMappingURL` comment
+   that corresponds to the privately hosted map server path.
+
+Then, when a developer needs to debug production code, they:
+
+1. Gain access to the private maps server. This could be, logging in to
+   the VPN, authenticating to a publicly-accessible server, firing up an SSH
+   tunnel to a non-public server, etc.
+
+After these steps, privately hosted source maps will be available for the
+production code.
+
+Advantages:
+
+* Developers can just "go online" and magically have the correct maps.
+  No checking out versions, rebuilding, etc.
+
+Disadvantages:
+
+* Need to publish to private servers during build process.
+* Need to host / control private map servers.
+
+##### Source Map Example
+
+(Example build / dist code available at: [github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-source-maps](https://github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-source-maps))
+
+Our Webpack configuration uses the
+[`SourceMapDevToolPlugin`](https://webpack.github.io/docs/list-of-plugins.html#sourcemapdevtoolplugin)
+directly for maximum flexibility. Note however that there are many catch-all
+[`devtool` options](https://webpack.github.io/docs/configuration.html#devtool)
+for simpler / development configuration. In our case here, we:
+
+- Enable minification with the `UglifyJsPlugin` so that we have a good example
+  to see the impact of source maps with.
+- Override the `append` option to `SourceMapDevToolPlugin` so that we can insert
+  a full URL in the outputted `//# sourceMappingURL` comment.
+
+[`webpack.config.js`](../../examples/frontend/webpack-source-maps/webpack.config.js)
+
+```js
+var path = require("path");
+var webpack = require("webpack");
+
+module.exports = {
+  context: path.join(__dirname, "../src/es5"),
+  entry: {
+    app1: "./app1.js",
+    app2: "./app2.js"
+  },
+  output: {
+    path: path.join(__dirname, "dist/js"),
+    filename: "[name].js"
+  },
+  plugins: [
+    new webpack.optimize.UglifyJsPlugin({
+      // Preserve the source map comment in minified code.
+      sourceMap: true
+    }),
+    new webpack.SourceMapDevToolPlugin({
+      // https://webpack.github.io/docs/configuration.html#output-sourcemapfilename
+      filename: "[name].js.map",
+
+      // Output sourceMappingURL comment.
+      //
+      // Examples:
+      // 1. Default to: `"\n//# sourceMappingURL=[url]"` for relative hosting
```

Comment:
i'm a little confused as to what `[url]` is? i assume it ends with `filename` as defined above. And then just the path to the file?